lib/elixir/lib/registry.ex

```diff
@@ -1318,6 +1318,47 @@ defmodule Registry do
     end
   end
 
+  @doc """
+  Works like `select/2`, but only returns the number of matching records.
+
+  ## Examples
+
+  In the example below we register the current process under different
+  keys in a unique registry but with the same value:
+
+      iex> Registry.start_link(keys: :unique, name: Registry.CountSelectTest)
+      iex> {:ok, _} = Registry.register(Registry.CountSelectTest, "hello", :value)
+      iex> {:ok, _} = Registry.register(Registry.CountSelectTest, "world", :value)
+      iex> Registry.count_select(Registry.CountSelectTest, [{{:_, :_, :value}, [], [true]}])
+      2
+  """
+  @spec count_select(registry, spec) :: non_neg_integer()
+  def count_select(registry, spec)
+      when is_atom(registry) and is_list(spec) do
+    spec =
+      for part <- spec do
+        case part do
+          {{key, pid, value}, guards, select} ->
+            {{key, {pid, value}}, guards, select}
+
+          _ ->
+            raise ArgumentError,
+                  "invalid match specification in Registry.select_count/2: #{inspect(spec)}"
+        end
+      end
```

Comment:
Please move this to a private function that can be reused by both select and select_count.

Reply:
Attemptted to do that here https://github.com/elixir-lang/elixir/pull/11405/commits/7d06e77047692b0c00881e0689e4c3e660fca23d 
Is there a better way to identify the function name of the caller than fetching it from `__ENV__.function`?

Reply:
This is good, thank you!